nixos/modules/services/networking/wireguard.nix

```diff
@@ -213,6 +227,48 @@ let
         '';
       };
 
+      endpointReceiving = {
+        enable = mkEnableOption ''
+          receiving the enpoint of other peers known by this peer,
+          which must have <option>networking.wireguard.peersAnnouncing</option> enabled
+          in its own configuration.
+
+          This is especially useful to punch-through a NAT
+          by establishing a direct peer-to-peer connectivity:
+          the peers behind a NAT initiate a connection
+          to the <option>networking.wireguard.listenPort</option>
+          of the announcing peer, they receive from this peer the endpoints to reach
+          the other peers behind a NAT, and except for symmetric NAT,
+          UDP hole punching enables them to establish a direct WireGuard tunnel between them.
+        '';
+        addr = mkOption {
+          type = types.str;
+          default = head (builtins.match "^\([^/]*\).*$" (head (config.allowedIPs)));
```

Comment:
This should be done via `mkDefault` in the `config` block, otherwise you'll have to rebuild the manual every time when one of these settings change. You may want to use `defaultText` to indicate that a default exists though.

Reply:
Here's a concrete example of how to do this, in the postgres module.

In the `options` section:

https://github.com/NixOS/nixpkgs/blob/6845a74d4c85e0a392ee82a829d7fc8c9ecf4f08/nixos/modules/services/databases/postgresql.nix#L67-L70

In the `config` section:

https://github.com/NixOS/nixpkgs/blob/6845a74d4c85e0a392ee82a829d7fc8c9ecf4f08/nixos/modules/services/databases/postgresql.nix#L300

Reply:
Sorry, I did know using `default` like this is a problem but forgot to fix it. Thanks!